### PR TITLE
Remove usage of deprecated distutils in cmake files

### PR DESCRIPTION
**Pull request by hmaarrfk**
I didn't have a chance to go through the whole repo, but I made this PR upstream to `zfpy` which vendors some files

https://github.com/LLNL/zfp/pull/217

If you want to monitor if this "works", please check build logs in https://github.com/conda-forge/zfpy-feedstock/pull/35

goodluck modernizing this.

There seems to be only a finite number of occurrences of distutils 

```
grep "(from|import) distutils" . -RE
```

### CODE REVIEW

**Comment by henryiii** [comment]
FYI, I think this might need to be conditional, as there were some issues with sysconfig before 3.10 or so.

**Comment by hmaarrfk** [comment]
lets see if i got the condition right.

**Comment by hmaarrfk** [comment]
Are you OK bumping cmake to 3.12?

If so, we can use

Python_SITELIB

https://cmake.org/cmake/help/latest/module/FindPython.html

**Comment by henryiii** [comment]
Ahh, forgot about this, feel free to ping me. This fixes #1047. I'd probably wait to increase the CMake version until scikit-build-core becomes the backend. And we can't move from FindPythonInterp/Libs to FindPython, that's a major breaking change - and people using FindPython don't really need this helper. (Also the first version that works correctly is 3.18-3.26 or so, depending on what you need. Which is why scikit-build-core provides an optional backport for 3.15+).

File: skbuild/resources/cmake/FindPythonExtensions.cmake [hunk 1]
```diff
@@ -254,19 +254,27 @@ endif()
 include(targetLinkLibrariesWithDynamicLookup)
 
 set(_command "
-import distutils.sysconfig
+import sys
+at_least_python_3_10 = sys.version_info[:2] >= (3, 10)
+if at_least_python_3_10:
```
Comment (henryiii) [inline skbuild/resources/cmake/FindPythonExtensions.cmake:259]:
```suggestion
if sys.version_info >= (3, 10):
```

This is an idiom already.

File: skbuild/resources/cmake/FindPythonExtensions.cmake [hunk 2]
```diff
@@ -254,19 +254,27 @@ endif()
...
 import site
-import sys
 
 result = None
 rel_result = None
 candidate_lists = []
 
 try:
-    candidate_lists.append((distutils.sysconfig.get_python_lib(),))
+    if at_least_python_3_10:
```
Comment (henryiii) [inline skbuild/resources/cmake/FindPythonExtensions.cmake:274]:
```suggestion
    if sys.version_info >= (3, 10):
```

File: skbuild/resources/cmake/FindPythonExtensions.cmake [hunk 3]
```diff
@@ -287,13 +295,17 @@ for candidate in candidates:
         rel_result = rel_candidate
         break
 
+if at_least_python_3_10:
```
Comment (henryiii) [inline skbuild/resources/cmake/FindPythonExtensions.cmake:298]:
```suggestion
if sys.version_info >= (3, 10):
```

**Comment by henryiii** [comment]
This produces:

```
Could NOT find NumPy (missing: NumPy_CONV_TEMPLATE_EXECUTABLE
  NumPy_FROM_TEMPLATE_EXECUTABLE) 
```

IMO NumPy changes should be broken out.

**Comment by hmaarrfk** [comment]
Sorry for the delay on my part, i was circling around to this since I remembered your suggestions and was surprised to see that you had merged them in, i have no problem with this. i decided to rebase and push.

One final thing: I want to make sure that you are still interested in Python 3.9 in light of [SPEC 0](https://scientific-python.org/specs/spec-0000/)

**Comment by henryiii** [comment]
@LecrisUT: Failure during Ansible playbook execution showing up for two fedora tests.

**Comment by LecrisUT** [comment]
> @LecrisUT: Failure during Ansible playbook execution showing up for two fedora tests.

Yeah, python 3.13 got merged 1-2 days ago and packages are still catching up. Give it 1-2 week

**Pull request merged**